docs/proposal/AI-gateway.md

```diff
@@ -0,0 +1,156 @@
+
+So we finally chose Envoy's [External Processing Filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_proc_filter). It connects to an external service, called an "external processor", to the filter chain. The processing service itself implements a gRPC interface that allows it to respond to events in the lifecycle of an HTTP request/response by examining and modifying the headers, body, and trailers of each message, or by returning a brand-new response.
+
+In this way, our AI plugin can be built as an external independent service and can be deployed on demand when AI traffic needs to be processed. At the same time, we can use any dev language, not necessarily C++. Of course, we use Golang. Also it is isolated from the complexity of Envoy to greatest extent, and only needs to process traffic requests/responses from gRPC connections. This fully ensures that development is simple, flexible and friendly.
+
+#### Integration with Existing Gateways
+
+The integration of various existing gateways and AI plugins is also very convenient. For example, Envoy Gateway provides a dedicated CRD for External Processing. As follows:
+
+```yaml
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: EnvoyExtensionPolicy
+metadata:
+  name: ext-proc-example
+  namespace: ollama
+spec:
+  targetRefs:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      name: ollama
+  extProc:
+  - backendRefs:
+    - name: aiengine
+      port: 9002
+    processingMode:
+      request:
+        body: Buffered
+      response:
+        body: Streamed
+
+```
+
+Istio Ingress Gateway can also use EnvoyFilter to add External Processing Filters on demand. Also, if necessary, as shown in the above arch picture, Envoy can be directly connected to AI plugin (it may no longer be appropriate to call it AI plugin), and then the AI plugin connects to the control plane of Gateway. The AI plugin modifies the XDS obtained from the control plane, for example, currently adding External Processing Filters, and finally sends the modified XDS to Envoy. This further improves the Gateway-agnosticity of the AI plugin and no additional configuration is required for different Gateway implementations.
+
+
+#### AI Plugin CRD
+
+Finally, we need to rely on CRD to describe the config related to AI traffic management. The config consists of the following two parts:
+
+1. Associated Service: The AI model is ultimately deployed in a service. We hope that only the service configured with the corresponding AI policy will forward traffic to the AI plugin and all AI services in the cluster can share the same AI plugin deployment without having to deploy one for each Service. In addition, if a service configured with AI policy, traffic can be directly forwarded to the AI plugin. Whether it's through automatically generating resource objects such as `EnvoyExtensionPolicy` or modifying XDS. It's believed that it will bring a better user experience.

```

Comment:
`Its believed that it will bring a better user experience.`  what does `it` really mean

Reply:
`it` means that we automatically insert envoy `external processing filter` to direct traffic to AI Plugin for services configured with AI policies, without users to explicitly configure it. For example, for `Envoy Gateway`, there is no need to explicitly configure `EnvoyExtensionPolicy`.